ci-operator/step-registry/cucushift/upgrade/tolatest/cucushift-upgrade-tolatest-commands.sh

```diff
@@ -0,0 +1,681 @@
+    TARGET_VERSION=$(cat "${tmp_log}"|awk '{print $NF}')
+    if [ -z "${TARGET_VERSION}" ] ; then
+        echo "Fail to get target version!"
+        exit 1
+    fi
+    export TARGET_VERSION
+    echo -e "Target release version is: ${TARGET_VERSION}\n"
+}
+
+# Monitor the upgrade status
+function check_upgrade_status() {
+    local wait_upgrade="${TIMEOUT}" out avail progress
+    echo "Starting the upgrade checking on $(date "+%F %T")"
+    while (( wait_upgrade > 0 )); do
+        sleep 5m
+        wait_upgrade=$(( wait_upgrade - 5 ))
+        if ! ( run_command "oc get clusterversion" ); then
+            continue
+        fi
+        if ! out="$(oc get clusterversion --no-headers)"; then continue; fi
+        avail="$(echo "${out}" | awk '{print $3}')"
+        progress="$(echo "${out}" | awk '{print $4}')"
+        if [[ ${avail} == "True" && ${progress} == "False" && ${out} == *"Cluster version is ${TARGET_VERSION}" ]]; then
+            echo -e "Upgrade succeed on $(date "+%F %T")\n\n"
+            return 0
+        fi
+    done
+    if [[ ${wait_upgrade} -le 0 ]]; then
+        echo -e "Upgrade timeout on $(date "+%F %T"), exiting\n" && return 1
+    fi
+}
+
+# Check version, state in history
+function check_history() {
+    local version state
+    version=$(oc get clusterversion/version -o jsonpath='{.status.history[0].version}')
+    state=$(oc get clusterversion/version -o jsonpath='{.status.history[0].state}')
+    if [[ ${version} == "${TARGET_VERSION}" && ${state} == "Completed" ]]; then
+        echo "History check PASSED, cluster is now upgraded to ${TARGET_VERSION}" && return 0
+    else
+        echo >&2 "History check FAILED, cluster upgrade to ${TARGET_VERSION} failed, current version is ${version}, exiting" && return 1
+    fi
+}
+
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# Get the target upgrades release, by default, OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE is the target release
+# If it's serial upgrades then override-upgrade file will store the release and overrides OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE
+# upgrade-edge file expects a comma separated releases list like target_release1,target_release2,...
+export TARGET_RELEASES=("${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}")
+if [[ -f "${SHARED_DIR}/upgrade-edge" ]]; then
+    release_string="$(< "${SHARED_DIR}/upgrade-edge")"
+    # shellcheck disable=SC2207
+    TARGET_RELEASES=($(echo "$release_string" | tr ',' ' '))
+fi
+echo "Upgrade targets are ${TARGET_RELEASES[*]}"
+
+export OC="run_command_oc"
+
+# Target version oc will be extract in the /tmp/client directory, use it first
+mkdir -p /tmp/client
+export OC_DIR="/tmp/client"
+export PATH=${OC_DIR}:$PATH
+
+for target in "${TARGET_RELEASES[@]}"
```

Comment:
For this step - `cucushift-upgrade-to` step, it should be only used in one-hop upgrade, so I think we can remove the loop, once that, we can also can remove the related codes about health check in this script, leave the health check to `cucushift-installer-check-cluster-health`, once that, we can largely shorten the code in this script.

Reply:
ok, let me update and limit it the one-hop upgrade only.